src/debug/ee/functioninfo.cpp

```diff
@@ -1616,15 +1610,21 @@ DebuggerJitInfo *DebuggerMethodInfo::FindOrCreateInitAndAddJitInfo(MethodDesc* f
 
     // The DJI may already be populated in the cache, if so CreateInitAndAddJitInfo is a no-op and that is fine.
     // CreateInitAndAddJitInfo takes a lock and checks the list again, which makes this thread-safe.
+
+    CodeVersionManager::TableLockHolder lockHolder(fd->GetCodeVersionManager());
+    CodeVersionManager *pCodeVersionManager = fd->GetCodeVersionManager();
+    NativeCodeVersion nativeCodeVersion = pCodeVersionManager->GetNativeCodeVersion(fd, startAddr);
+    _ASSERTE(!nativeCodeVersion.IsNull());
```

Comment:
Unforetunately EnC doesn't currently play nice with code versioning. If you edit EnC methods no new NativeCodeVersion gets generated and this search will return NULL. In that case you should be fine using the default NativeCodeVersion:

```
//some day we'll get EnC to use code versioning properly, but until then we'll get the right behavior treating all EnC versions as the default native code version.
if(nativeCodeVersion.IsNull())
{
    nativeCodeVersion = NativeCodeVersion(fd)
}
```